detection_rules/packaging.py

```diff
@@ -149,13 +150,15 @@ def manage_versions(rules: list, deprecated_rules: list = None, current_versions
 class Package(object):
     """Packaging object for siem rules and releases."""
 
-    def __init__(self, rules, name, deprecated_rules=None, release=False, current_versions=None, min_version=None,
-                 max_version=None, update_version_lock=False, verbose=True):
+    def __init__(self, rules: List[Rule], name, deprecated_rules: List[Rule] = None, release=False,
+                 current_versions: dict = None, min_version: int = None, max_version: int = None,
+                 update_version_lock=False, registry_data: dict = None, verbose=True):
```

Comment:
if you want a type to be X or None, then you  should do `Optional[None]`.
assuming you add `from typing import Optional` to the file:

```suggestion
    def __init__(self, rules: List[Rule], name, deprecated_rules: List[Rule] = None, release=False,
                 current_versions: Optional[dict] = None, min_version: Optional[int] = None, max_version: Optional[int] = None,
                 update_version_lock=False, registry_data: Optional[dict] = None, verbose=True):
```

Reply:
Thanks for catching this - will expand typing completely for all these args as well